backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/GeneralViewDiagramDescriptionProvider.java

```diff
@@ -676,7 +678,12 @@ private List<NodeTool> getCreationToolsForElement(EClass element, String toolSec
         List<NodeTool> elementCreationTools = new ArrayList<>();
         NodeDescription nodeDescription = cache.getNodeDescription(this.getDescriptionNameGenerator().getNodeName(element)).orElse(null);
         if (nodeDescription != null) {
-            NodeTool nodeTool = this.toolDescriptionService.createNodeToolFromDiagramBackground(nodeDescription, element);
+            NodeTool nodeTool = null;
+            if (SysmlPackage.eINSTANCE.getNamespaceImport().equals(element)) {
```

Comment:
Why are you doing this?
It seems strange to this "patch" of this generic code, only for your new tool.
There are other existing tools involving SelectionDialog, and there's no need to do that here for those tools.

Reply:
Changed with `addCustomNodes` mechanism